grakn-core/src/main/java/ai/grakn/graql/Var.java

```diff
@@ -109,84 +120,98 @@
      * @param var a variable representing a resource
      * @return this
      */
+    @CheckReturnValue
     Var has(TypeName type, Var var);
 
     /**
      * @param type a concept type id that the variable must be of this type
      * @return this
      */
+    @CheckReturnValue
     Var isa(String type);
 
     /**
      * @param type a concept type that this variable must be an instance of
      * @return this
      */
+    @CheckReturnValue
     Var isa(Var type);
 
     /**
      * @param type a concept type id that this variable must be a kind of
      * @return this
      */
+    @CheckReturnValue
     Var sub(String type);
 
     /**
      * @param type a concept type that this variable must be a kind of
      * @return this
      */
+    @CheckReturnValue
     Var sub(Var type);
 
     /**
      * @param type a role type id that this relation type variable must have
      * @return this
      */
+    @CheckReturnValue
     Var hasRole(String type);
 
     /**
      * @param type a role type that this relation type variable must have
      * @return this
      */
+    @CheckReturnValue
     Var hasRole(Var type);
 
     /**
      * @param type a role type id that this concept type variable must play
      * @return this
      */
+    @CheckReturnValue
     Var playsRole(String type);
 
     /**
      * @param type a role type that this concept type variable must play
      * @return this
      */
+    @CheckReturnValue
     Var playsRole(Var type);
 
     /**
      * @param type a scope that this variable must have
      * @return this
      */
+    @CheckReturnValue
     Var hasScope(Var type);
 
     /**
      * @param type a resource type that this type variable can be related to
      * @return this
      */
+    @CheckReturnValue
     Var hasResource(String type);
 
     /**
      * @param type a resource type that this type variable can be related to
      * @return this
      */
+    @CheckReturnValue
     Var hasResource(Var type);
 
     /**
      * @param type a resource type that this type variable can be one-to-one related to
      * @return this
      */
+    @CheckReturnValue
```

Comment:
It's a suggestion that findbugs recognises, stating that a caller to this function should use the return value for something.

Because `Var` is immutable, calling these functions will never have any side-effects, so if a caller does not use the return value, then they have made a mistake. This helped findbugs catch several places in our code where we assumed `Var` was mutable!